ZeldaFullEditor/ZSCustomOverworld.asm

```diff
@@ -2835,33 +3134,72 @@ ReplaceBGColor:
 {
     PHB : PHK : PLB
 
-    ; TODO: This may need to check if we are in a warp and then if so load the
-    ; custom color. If not, then chceck if its enabled or not.
-    ;SEP #$20 ; Set A in 8bit mode.
+    SEP #$20 ; Set A in 8bit mode.
 
-    ;LDA.w Pool_EnableBGColor : BNE .custom
-        ;REP #$20 ; Set A in 16bit mode.
+    LDA.w Pool_EnableBGColor : BNE .custom
+        REP #$20 ; Set A in 16bit mode.
 
-        ;PLB
+        PLB
 
-        ;RTL
+        RTL
 
-    ;.custom
+    .custom
 
-    ;REP #$20 ; Set A in 16bit mode.
+    REP #$20 ; Set A in 16bit mode.
 
-    LDA.b $8A : ASL : TAX ; Get area code and times it by 2.
-    LDA.w Pool_BGColorTable, X ; Get the color.
+    ; TODO: Saving the Y may not be necessary, done just in case.
+    PHY
 
-    ;STA.l $7EC300 : STA.l $7EC340 ; Set the BG color.
+    ; Get area code and times it by 2.
+    LDA.b $8A : ASL : TAX
+
+    ; Get the color.
+    LDA.w Pool_BGColorTable, X
+
+    ; Set the BG color.
     STA.l $7EC500 : STA.l $7EC540
 
+    TAY ; Save the color.
+    
+    SEP #$20 ; Set A in 8bit mode.
+
+    ; TODO: Pretty sure this is needed. Just keep an eye out for it.
+    ; Set the buffer color when exiting to the OW to prevent a bug when using the
+    ; map in an area with a subscreen overlay.
+    LDA.b $10 : CMP.b #$08 : BNE .notPreOverworld
+        BRA .setBuffer
+
+    .notPreOverworld
+
+    ; TODO: Check if this is needed. I think it is. If not, it should always
```

Comment:
I will definitely investigate this further when I test this with the Oracle Day/Night system.

Reply:
sweet